snap/info.go

```diff
@@ -643,28 +643,43 @@ func (s *Info) CommonDataSaveDir() string {
 	return CommonDataSaveDir(s.InstanceName())
 }
 
-// DataHomeGlob returns the globbing expression for the snap directories in use
-func DataHomeGlob(opts *dirs.SnapDirOptions) string {
+// DataHomeGlobs returns a slice of globbing expressions for the snap directories in use
+func DataHomeGlobs(opts *dirs.SnapDirOptions) []string {
 	if opts == nil {
 		opts = &dirs.SnapDirOptions{}
 	}
 
-	if opts.HiddenSnapDataDir {
-		return dirs.HiddenSnapDataHomeGlob
+	list := make([]string, len(dirs.GetSnapHomeDirs()))
+	copy(list, dirs.GetSnapHomeDirs())
```

Comment:
thank you, this very clever. great catch avoiding mutating the slice accidentally. would you mind moving making a copy inside `GetSnapHomeDirs()` to make sure that all usages are safe and don't mutate the original slice.

Reply:
Given that this is the only usage of `GetSnapHomeDirs`, you could ignore my comment. but it won't hurt to be safe.

Reply:
I agree it makes much more sense to have this in the `GetSnapHomeDirs` function